synedrion/src/cggmp21/protocols/presigning.rs

```diff
@@ -344,25 +350,41 @@ impl<P: SchemeParams, I: Debug + Clone + Ord + Serialize> Round<I> for Round2<P,
         let aux = (&self.context.ssid_hash, &self.my_id());
 
         let cap_gamma = self.context.gamma.mul_by_generator();
-        let pk = &self.context.aux_info.secret_aux.paillier_sk.public_key();
+        let pk = self.context.aux_info.secret_aux.paillier_sk.public_key();
 
         let target_pk = &self.context.aux_info.public_aux[destination].paillier_pk;
 
-        let beta = Signed::random_bounded_bits(rng, P::LP_BOUND);
-        let hat_beta = Signed::random_bounded_bits(rng, P::LP_BOUND);
-        let r = RandomizerMod::random(rng, pk);
-        let s = RandomizerMod::random(rng, target_pk);
-        let hat_r = RandomizerMod::random(rng, pk);
-        let hat_s = RandomizerMod::random(rng, target_pk);
+        let beta = Signed::random_bounded_bits(rng, P::LP_BOUND).secret_box();
+        let hat_beta = Signed::random_bounded_bits(rng, P::LP_BOUND).secret_box();
+        let r = RandomizerMod::random(rng, pk).secret_box();
+        let s = RandomizerMod::random(rng, target_pk).secret_box();
+        let hat_r = RandomizerMod::random(rng, pk).secret_box();
+        let hat_s = RandomizerMod::random(rng, target_pk).secret_box();
 
-        let cap_f = CiphertextMod::new_with_randomizer_signed(pk, &beta, &r.retrieve());
+        let cap_f = CiphertextMod::new_with_randomizer_signed(
+            pk,
+            beta.expose_secret(),
+            r.expose_secret().retrieve().secret_box(),
+        );
         let cap_d = &self.all_cap_k[destination] * P::signed_from_scalar(&self.context.gamma)
-            + CiphertextMod::new_with_randomizer_signed(target_pk, &-beta, &s.retrieve());
+            + CiphertextMod::new_with_randomizer_signed(
+                target_pk,
+                &-beta.expose_secret(),
+                s.expose_secret().retrieve().secret_box(),
```

Comment:
There are a few of these "expose-modify-wrap-again" that made me wish there was a more powerful API offered by `secrecy` to let me act on the wrapped secret and return the result as a new `SecretBox`, something like `s.transform_inner(|secret| { secret.retrieve() }`. But perhaps the awkwardness is a feature here?